### PR TITLE
fix(application): use correct attribute typehints

### PR DESCRIPTION
**Pull request by ooliver1**
## Description

- `Optional` for flags as its ommitable
- `[item]` for when it is "guaranteed"
- `Class([item]) if item in data` for when it is a nullable/ommittable object
- `List[str]` to represent the type of items

## Checklist

- [ ] - All code changes have been tested, if any
- [ ] - The documentation has been updated accordingly for any code changes
- [ ] - This PR fixes an already existing issue
- [ ] - This PR adds a new feature (new functions, methods, classes, parameters, etc.)
- [ ] - This PR fixes a bug
- [ ] - This PR is a breaking change (removes/renames functions, parameters etc.)
- [ ] - This PR is NOT a code change (changes Documentation files, README.md, CONTRIBUTING.md etc.)

### CODE REVIEW

File: EpikCord/application/application.py [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
         self.terms_of_service_url: Optional[str] = data.get("terms_of_service")
         self.privacy_policy_url: Optional[str] = data.get("privacy_policy")
         self.owner: Optional[PartialUser] = (
-            PartialUser(data.get("user")) if data.get("user") else None
+            PartialUser(data["user"]) if "user" in data else None
         )
-        self.verify_key: str = data.get("verify_key")
-        self.team: Optional[Team] = Team(data.get("team")) if data.get("get") else None
+        self.verify_key: str = data["verify_key"]
+        self.team: Optional[Team] = Team(data["team"]) if data["team"] else None
```
Comment (TheUntraceable) [inline EpikCord/application/application.py:22]: Wouldn't this return a KeyError?

**Review by TheUntraceable — APPROVED** [review]
If you just answer me in the comment I left and changes are made where needed I'll merge

File: EpikCord/application/application.py (hunk 1 above)
Comment (ooliver1) [reply EpikCord/application/application.py:22]: image

The `?` is on the type which means it is nullable (`T | None`).
This creates a `Team` object if the `team` field is not `None`

Comment (TheUntraceable) [reply EpikCord/application/application.py:22]: Yes but the `if date["team"] else None` would raise a KeyError
```py
>>> data = {} # The key doesn't exist
>>> bar = int(data["foo"]) if data["foo"] else None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foo'```

Comment (ooliver1) [reply EpikCord/application/application.py:22]: What I'm showing you is `?type` means the key **will exist**, but will just be **nullable**.

Comment (TheUntraceable) [reply EpikCord/application/application.py:22]: My bad, I thought that was about the key.

**Pull request merged**